Add eager for keras benchmark

yhliang2018 opened the pull request:
Hi All,

Could you help to review this PR? This PR is created to refine the keras model benchmark by:
1) Use models_helper for synthetic dataset generation
2) Add eager mode for keras model. 

Some notes about eager mode: to run with eager mode, the optimizer should be TF native; Also, currently eager mode and keras multi_gpu_model are not compatible. With eager mode enabled, only one GPU is used even multiple GPUs are provided.

Thanks,
Yanhui

karmel commented on official/keras_application_models/benchmark_main.py:
```diff
@@ -54,6 +58,10 @@ def run_keras_model_benchmark(_):
     raise AssertionError("The --model command line argument should "
                          "be a key in the `MODELS` dictionary.")
 
+  # Check if eager mode is enabled
```
Can we add the caveats you mentioned in the PR message to the comments somewhere too? Maybe to the eager flag help.

karmel commented on official/keras_application_models/benchmark_main.py:
```diff
@@ -46,6 +46,10 @@
 
+# Number of synthetic images as the same number of images in ImageNet
+_IMAGENET_TRAIN_IMAGES = 1281167
+_IMAGENET_EVAL_IMAGES = 50000
```
I wouldn't bother making this complicated for synthetic data. We can re-evaluate if we add real data processing, but for now, maybe a single flag that defaults to 1000?

anj-s commented on official/keras_application_models/benchmark_main.py:
```diff
@@ -64,30 +71,31 @@ def run_keras_model_benchmark(_):
 
   # If run with multiple GPUs
+  # If eager mode is enabled, only one GPU is utilized even multiple GPUs
+  # are provided
```
Do you want to raise a warning or error message if eager execution is enabled and we use multi_gpu_model?

yhliang2018 replied:
Sure, I will add a warning message here. Then it keeps consistent with the help message in the --eager flag.

anj-s commented on official/keras_application_models/benchmark_main.py:
```diff
@@ -146,6 +154,18 @@ def define_keras_benchmark_flags():
+      name="eager", default=False, help=flags_core.help_wrap(
+          "To enable eager execution. Note that if eager execution is enabled, "
+          "only one GPU is utilized even multiple GPUs are provided and "
+          "multi_gpu_model is used."))
```
s/even/even if

anj-s commented on official/keras_application_models/benchmark_main.py:
```diff
@@ -146,6 +154,18 @@ def define_keras_benchmark_flags():
+      name="num_images", default=_NUM_IMAGES,
+      help=flags_core.help_wrap(
+          "The number of synthetic images for training and evaluation. The "
+          "default value is 1000."))
```
Maybe I missed something, but is this the only place this constant is used? If so then do we just want the default to be specified here?

yhliang2018 replied:
Yeah, we use 1000 as the default number of images for both train and eval with synthetic dataset. Will just put a value here.

yhliang2018 commented:
Hi Anjali and Karmel,

I have addressed all the comments. The benchmark is already set up. As long as this PR is approved, we should be able to get some benchmark numbers, like exp/sec.

Thanks,
Yanhui

karmel commented on official/keras_application_models/dataset.py:
```diff
@@ -19,6 +19,9 @@
 
 import tensorflow as tf
 
+# pylint: disable=g-bad-import-order
```
This should be at the end of the line, or re-enabled after the relevant section
